### PR TITLE
misc: fix parameter types in MPI_Info_create_env

### LINKED ISSUE #5399: MPI-4: Bad C binding for MPI_Info_create_env()
The MPI-4 C binding for `MPI_Info_create_env` should be
```c
int MPI_Info_create_env(int argc, char argv[], MPI_Info *info)
```

I believe that is a typo, it should really be like a C `main` function argument list:
```c
int MPI_Info_create_env(int argc, char *argv[], MPI_Info *info)
```

In any case, in MPICH we currently have:
```c
int MPI_Info_create_env(int argc[], char ***argv, MPI_Info *info) MPICH_API_PUBLIC;
```

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
The parameters for MPI_Info_create_env should be:
```
int MPI_Info_create_env(int argc, char *argv[], MPI_Info *info)
```

Fixes #5399

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: maint/local_python/binding_common.py [hunk 1]
```diff
@@ -122,16 +122,31 @@ def get_C_param(param, mapping):
...
+    if kind == "ARGUMENT_COUNT":
+        if re.match(r'mpi_info_create_env', func['name'], re.IGNORECASE):
+            # -> int argc
+            pass
+        else:
+            # MPI_Init, MPI_Init_thread -> int *argc
+            want_star = 1
+    elif kind == "ARGUMENT_LIST":
+        if re.match(r'mpi_info_create_env', func['name'], re.IGNORECASE):
+            # -> char *argc[]
```
Comment (raffenet) [inline maint/local_python/binding_common.py:136]: should be `char *argv[]`

**Review by raffenet — COMMENTED** [review]
just one minor issue

File: maint/local_python/binding_common.py (hunk 1 above)
Comment (hzhou) [reply maint/local_python/binding_common.py]: Good catch!

Now fixed.

**Comment by dalcinl** [comment]
@hzhou @raffenet It looks like this one is ready to go.

**Review by raffenet — APPROVED** [review]

**Pull request merged**